File: main/src/main/java/lavaplayer/container/playlists/HlsStreamTrack.java
package lavaplayer.container.playlists;

import lavaplayer.source.stream.M3uStreamSegmentUrlProvider;
import lavaplayer.source.stream.MpegTsM3uStreamAudioTrack;
import lavaplayer.tools.io.HttpInterface;
import lavaplayer.tools.io.HttpInterfaceManager;
import lavaplayer.track.AudioTrackInfo;

public class HlsStreamTrack extends MpegTsM3uStreamAudioTrack {
    private final HlsStreamSegmentUrlProvider segmentUrlProvider;
    private final HttpInterfaceManager httpInterfaceManager;

    /**
     * @param trackInfo            Track info
     * @param httpInterfaceManager
     */
    public HlsStreamTrack(AudioTrackInfo trackInfo, String streamUrl, HttpInterfaceManager httpInterfaceManager,
                          boolean isInnerUrl) {

        super(trackInfo);

        segmentUrlProvider = isInnerUrl ?
            new HlsStreamSegmentUrlProvider(null, streamUrl) :
            new HlsStreamSegmentUrlProvider(streamUrl, null);

        this.httpInterfaceManager = httpInterfaceManager;
    }

    @Override
    protected M3uStreamSegmentUrlProvider getSegmentUrlProvider() {
        return segmentUrlProvider;
    }

    @Override
    protected HttpInterface getHttpInterface() {
        return httpInterfaceManager.getInterface();
    }
}

File: main/src/main/java/lavaplayer/container/playlists/HlsStreamTrack.kt
package lavaplayer.container.playlists

import lavaplayer.source.stream.MpegTsM3uStreamAudioTrack
import lavaplayer.tools.io.HttpInterface
import lavaplayer.tools.io.HttpInterfaceManager
import lavaplayer.track.AudioTrackInfo

/**
 * @param trackInfo            Track info
 * @param httpInterfaceManager The http interface manager
 */
class HlsStreamTrack(
    trackInfo: AudioTrackInfo,
    streamUrl: String,
    val httpInterfaceManager: HttpInterfaceManager,
    isInnerUrl: Boolean
) : MpegTsM3uStreamAudioTrack(trackInfo) {
    override val segmentUrlProvider: HlsStreamSegmentUrlProvider = if (isInnerUrl) {
        HlsStreamSegmentUrlProvider(null, streamUrl)
    } else {
        HlsStreamSegmentUrlProvider(streamUrl, null)
    }

    override val httpInterface: HttpInterface
        get() = httpInterfaceManager.get()
}
